Organize Sample Code

Vidushi-GitHub opened the pull request:
# Description
Re-organization of sample code is carried out as following:
-Kafka Messages
-XML
-JSON
-HEALPix

# Related Issue(s)
#2851

Vidushi-GitHub committed:
organize sample code

implemented feedbacks

feedback

text message page and save alert to Kafka

link error fix

Bump isbot from 5.1.23 to 5.1.26

Bumps [isbot](https://github.com/omrilotan/isbot) from 5.1.23 to 5.1.26.
- [Changelog](https://github.com/omrilotan/isbot/blob/main/CHANGELOG.md)
- [Commits](https://github.com/omrilotan/isbot/compare/v5.1.23...v5.1.26)

---
updated-dependencies:
- dependency-name: isbot
  dependency-version: 5.1.26
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Bump path-to-regexp and express

Bumps [path-to-regexp](https://github.com/pillarjs/path-to-regexp) and [express](https://github.com/expressjs/express). These dependencies needed to be updated together.

Updates `path-to-regexp` from 6.2.2 to 8.2.0
- [Release notes](https://github.com/pillarjs/path-to-regexp/releases)
- [Changelog](https://github.com/pillarjs/path-to-regexp/blob/master/History.md)
- [Commits](https://github.com/pillarjs/path-to-regexp/compare/v6.2.2...v8.2.0)

Updates `express` from 4.21.0 to 4.21.2
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/4.21.2/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.21.0...4.21.2)

---
updated-dependencies:
- dependency-name: path-to-regexp
  dependency-version: 8.2.0
  dependency-type: indirect
- dependency-name: express
  dependency-version: 4.21.2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Bump actions/download-artifact from 4.2.1 to 4.3.0

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 4.2.1 to 4.3.0.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/95815c38cf2ff2164869cbab79da8d1f422bc89e...d3f86a106a0bac45b974a628896c90dbdf5c8093)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-version: 4.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Bump isbot from 5.1.26 to 5.1.27

Bumps [isbot](https://github.com/omrilotan/isbot) from 5.1.26 to 5.1.27.
- [Changelog](https://github.com/omrilotan/isbot/blob/main/CHANGELOG.md)
- [Commits](https://github.com/omrilotan/isbot/compare/v5.1.26...v5.1.27)

---
updated-dependencies:
- dependency-name: isbot
  dependency-version: 5.1.27
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @nasa-gcn/schema from 4.2.0 to 4.3.2

Bumps [@nasa-gcn/schema](https://github.com/nasa-gcn/gcn-schema) from 4.2.0 to 4.3.2.
- [Release notes](https://github.com/nasa-gcn/gcn-schema/releases)
- [Commits](https://github.com/nasa-gcn/gcn-schema/compare/v4.2.0...v4.3.2)

---
updated-dependencies:
- dependency-name: "@nasa-gcn/schema"
  dependency-version: 4.3.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Bump esbuild from 0.25.2 to 0.25.3

Bumps [esbuild](https://github.com/evanw/esbuild) from 0.25.2 to 0.25.3.
- [Release notes](https://github.com/evanw/esbuild/releases)
- [Changelog](https://github.com/evanw/esbuild/blob/main/CHANGELOG.md)
- [Commits](https://github.com/evanw/esbuild/compare/v0.25.2...v0.25.3)

---
updated-dependencies:
- dependency-name: esbuild
  dependency-version: 0.25.3
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Circulars: Enable advanced search (#2994)

Update CHIME page

Bump vite

Bumps  and [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite). These dependencies needed to be updated together.

Updates `vite` from 6.2.6 to 6.3.4
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.3.4/packages/vite)

Updates `vite` from 4.5.13 to 6.3.4
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.3.4/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.3.4
  dependency-type: indirect
- dependency-name: vite
  dependency-version: 6.3.4
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Add sample code for PySpark

Fix file extension

remove references to Lucene where not appropriate

switches language in documentation to refer to search syntax as advanced search syntax rather than Lucene for consistency. We retain one reference to Lucene syntax when linking to the official Lucene documentation

adding eventId to edit props

Add FAQ entry about invalid token error message

Updated VTP migration documentation (#3080)

Vidushi-GitHub commented:
Style Reference:
<img width="907" alt="Screenshot 2025-02-21 at 4 20 48 PM" src="https://github.com/user-attachments/assets/fec0415f-145d-47f5-ae6e-ea4680c7d74d" />

Vidushi-GitHub commented:
Style reference:
<img width="953" alt="Screenshot 2025-02-21 at 4 22 20 PM" src="https://github.com/user-attachments/assets/0e9fbb9c-dbff-4d97-82c6-2036a5b6744a" />

Vidushi-GitHub commented on app/routes/docs.tsx:
```diff
@@ -162,7 +162,16 @@ export default function () {
                     </Link>,
                     <NavLink key="code-samples" to="client/samples">
-                      Sample Code
+                      Kafka Messages
```
Done

Vidushi-GitHub commented:
<img width="930" alt="Screenshot 2025-02-24 at 12 56 41 PM" src="https://github.com/user-attachments/assets/e0cf1c7f-aa50-4831-b368-668d3688a819" />

tylerbarna commented:
Very minor note, but maybe it would be better to have JSON/Healpix first? Also, I think the current Healpix doc is written more as a starter guide rather than a collection of sample code in the same way as the other two

jracusin commented:
Some of the sample code is just general kafka usage, and not specific to the message serialization.  There should be a separate section on sample code that's just general Kafka.

Vidushi-GitHub commented:
Except what is Kafka and how we stream data, content is refined as below:
<img width="871" alt="Screenshot 2025-03-14 at 5 03 42 PM" src="https://github.com/user-attachments/assets/90ae73e6-460e-4110-b2c4-058392f83649" />

Vidushi-GitHub commented:
<img width="962" alt="Screenshot 2025-03-19 at 2 11 27 PM" src="https://github.com/user-attachments/assets/09de6eca-4f69-45c0-bae2-a19e47f242fb" />

Vidushi-GitHub commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+Here is a collection of functions and example code that may be useful
+when working with GCN Notices. The sample code here demonstrates how to handle GCN Notices in both JSON and XML formats, and some samples from the FAQs section of the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
+
+### The following sections provide guidance on handling different GCN Notice formats:
```
```suggestion
### The following sections provide guidance on handling different GCN Notices formats:
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+
+# Apache Kafka and GCN Streaming
+
+[Apache Kafka](https://kafka.apache.org) is an open-source distributed event streaming platform used by thousands of companies for high-performance data pipelines, streaming analytics, data integration, and mission-critical applications.
```
```suggestion
[Apache Kafka](https://kafka.apache.org) is an open-source distributed event streaming platform used by GCN to stream Notices and Circulars in real time.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+
+[Apache Kafka](https://kafka.apache.org) is an open-source distributed event streaming platform used by thousands of companies for high-performance data pipelines, streaming analytics, data integration, and mission-critical applications.
+
+NASA leverages Kafka extensively, with notable implementations including the General Coordinates Network (GCN) at the Goddard Space Flight Center, the Complex Event Processor for Deep Space Network at Jet Propulsion Laboratory, and other federal agencies. These organizations use self-managed Kafka brokers, either Apache Kafka or the Confluent Platform, to support their data-intensive operations.  ...
```
```suggestion
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+
+[Apache Kafka](https://kafka.apache.org) is an open-source distributed event streaming platform used by thousands of companies for high-performance data pipelines, streaming analytics, data integration, and mission-critical applications.
+
+NASA leverages Kafka extensively, with notable implementations including the General Coordinates Network (GCN) at the Goddard Space Flight Center, the Complex Event Processor for Deep Space Network at Jet Propulsion Laboratory, and other federal agencies. These organizations use self-managed Kafka brokers, either Apache Kafka or the Confluent Platform, to support their data-intensive operations.  ...
+
+GCN’s Kafka cluster is simple yet flexible, it operates on a standard three-broker Kafka cluster, without any custom authentication extensions or vendor lock-in. It uses Confluent Platform but can easily switch to open-source Kafka or managed solutions like Confluent Cloud. With [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) for single sign-on, utilizing [Amazon Cognito ...
```
```suggestion
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+
+GCN Notices provide real-time machine readable alerts and are distributed in different formats:
+
+- New GCN Notices are provided in JSON format and are streamed via Kafka.
```
```suggestion
- New GCN Notices are provided in JSON format (GCN Unified Schema) and are streamed via Kafka.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+GCN Notices provide real-time machine readable alerts and are distributed in different formats:
+
+- New GCN Notices are provided in JSON format and are streamed via Kafka.
+- Classic GCN Notices are formatted in XML as well, are distributed both via Kafka and email.
```
```suggestion
- GCN Classic Notices are formatted in text, binary and VOEvent XML, are distributed both via Kafka and email.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,33 @@
+## Sample Code
+
+Here is a collection of functions and example code that may be useful
+when working with GCN Notices. The sample code here demonstrates how to handle GCN Notices in both JSON and XML formats, and some samples from the FAQs section of the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
```
```suggestion
when working with GCN Notices. This sample code demonstrates how to handle GCN Notices in both JSON and VOEvent XML formats, and some samples from the FAQs section of the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
```

Vidushi-GitHub commented:
Screenshot of recent version
<img width="908" alt="Screenshot 2025-03-28 at 3 50 33 PM" src="https://github.com/user-attachments/assets/ef0bdafc-149b-4159-971e-1dbfb7ed4a0e" />

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,30 @@
+
+GCN Notices provide real-time machine readable alerts and are distributed in different formats:
+
+- New GCN Notices are provided in JSON format and are streamed via Kafka. Note SVOM available via VOEvent over Kafka.
```
```suggestion
- New GCN Notices are provided in JSON format and are streamed via Kafka. Note that SVOM notices are only available via VOEvent over Kafka and email.
```

jracusin replied:
[Adding email for SVOM](https://github.com/nasa-gcn/gcn-email/pull/281)

Vidushi-GitHub replied:
> [Adding email for SVOM](https://github.com/nasa-gcn/gcn-email/pull/281)

Shall we wait on this?
I am committing suggestions.

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,30 @@
+GCN Notices provide real-time machine readable alerts and are distributed in different formats:
+
+- New GCN Notices are provided in JSON format and are streamed via Kafka. Note SVOM available via VOEvent over Kafka.
+- GCN Classic Notices are formatted in text, binary and VOEvent XML, are distributed both via Kafka and email.
```
```suggestion
- GCN Classic Notices are provided in text, binary and VOEvent XML, and are distributed both via Kafka and email.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,30 @@
+- [JSON Data](sample/json) provides sample code for handling JSON-based Notices
+- [HEALPix](sample/healpix) explains how to work with HEALPix sky maps
+
+To contribute your own ideas, make a GitHub pull request to add it to [the Markdown source for this document](https://github.com/nasa-gcn/gcn.nasa.gov/blob/CodeSamples/app/routes/docs.sample._index.mdx), or [contact us](/contact).
```
```suggestion
To contribute your own sample code, make a GitHub pull request to add it to [the Markdown source for this document](https://github.com/nasa-gcn/gcn.nasa.gov/blob/CodeSamples/app/routes/docs.sample._index.mdx), or [contact us](/contact).
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,30 @@
+  breadcrumb: Sample Code
+---
+
+# Apache Kafka and GCN Streaming
```
This shouldn't be the level 1 header.  The main header should "Sample Code".

Vidushi-GitHub commented:
Screenshot 
<img width="981" alt="Screenshot 2025-04-01 at 4 21 47 PM" src="https://github.com/user-attachments/assets/96b383db-3e4e-42a9-851b-c7798076f703" />

lpsinger commented on app/routes/docs.client.samples.md:
Please add a redirect to the new location of this content.

tylerbarna commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,28 @@
+- New GCN Notices are provided in JSON format and are streamed via Kafka. Note that SVOM notices are only available via VOEvent over Kafka and email.
+- GCN Classic Notices are provided in text, binary and VOEvent XML, and are distributed both via Kafka and email.
+
+## The following sections provide guidance on handling different GCN Notices formats:
```
```suggestion
## Additional Information
The following sections provide guidance on handling different GCN Notices formats:
```

tylerbarna commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,28 @@
+
+## The following sections provide guidance on handling different GCN Notices formats:
+
+- [Kafka Usage](sample/kafka) explains Kafka's functionality
```
```suggestion
- [Kafka Consumption](sample/kafka) explains Kafka's functionality and how to consume notices.
```

Vidushi-GitHub replied:
I have modified, please check latest.

tylerbarna commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,28 @@
+## The following sections provide guidance on handling different GCN Notices formats:
+
+- [Kafka Usage](sample/kafka) explains Kafka's functionality
+- [XML Data](sample/xml) guides users for working with XML Notices
```
```suggestion
- [XML Data](sample/xml) guides users on working with XML Notices.
```

tylerbarna commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,28 @@
+
+- [Kafka Usage](sample/kafka) explains Kafka's functionality
+- [XML Data](sample/xml) guides users for working with XML Notices
+- [JSON Data](sample/json) provides sample code for handling JSON-based Notices
```
```suggestion
- [JSON Data](sample/json) provides sample code for handling JSON-based Notices.
```

tylerbarna commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,28 @@
+- [Kafka Usage](sample/kafka) explains Kafka's functionality
+- [XML Data](sample/xml) guides users for working with XML Notices
+- [JSON Data](sample/json) provides sample code for handling JSON-based Notices
+- [HEALPix](sample/healpix) explains how to work with HEALPix sky maps
```
```suggestion
- [HEALPix](sample/healpix) explains how to work with HEALPix sky maps.
```

tylerbarna commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,29 @@
+Here is a collection of functions and example code that may be useful
+when working with GCN Notices. This sample code demonstrates how to handle GCN Notices in both JSON and VOEvent XML formats, and some samples from the FAQs section of the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
+
+## Apache Kafka and GCN Streaming
+
+[Apache Kafka](https://kafka.apache.org) is an open-source distributed event streaming platform used by GCN to stream Notices and Circulars in real time.
+
+## GCN Notices Formats and Distributions
+
+GCN Notices provide real-time machine readable alerts and are distributed in different formats:
+
+- New GCN Notices are provided in JSON format and are streamed via Kafka. Note that SVOM notices are only available via VOEvent over Kafka and email.
+- GCN Classic Notices are provided in text, binary and VOEvent XML, and are distributed both via Kafka and email.
```
Perhaps we could clarify the distribution methods available for the two types on the main notices docs page and then link to it here?

tylerbarna reviewed (changes requested):
see above

Vidushi-GitHub commented on app/routes/docs.sample.ts:
```diff
@@ -0,0 +1,16 @@
+
+export const loader: LoaderFunction = async () => {
+  return redirect(
+    'https://github.com/nasa-gcn/gcn.nasa.gov/blob/CodeSamples/app/routes/docs.sample._index.mdx',
```
@dakota002 not sure with correct url link

jracusin commented:
> see above

#2884 should help address this, but please do link to the Notices docs

Vidushi-GitHub commented:
<img width="793" alt="Screenshot 2025-04-21 at 1 58 50 PM" src="https://github.com/user-attachments/assets/b82fadb1-ffec-4d11-bf8e-9fa923886bba" />

Vidushi-GitHub commented:
> > see above
> 
> #2884 should help address this, but please do link to the Notices docs

I have linked to notices docs page.

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,28 @@
+
+- New GCN Notices are provided in JSON format and are streamed via Kafka. Note that SVOM notices are only available via VOEvent over Kafka and email.
+- GCN Classic Notices are provided in text, binary and VOEvent XML, and are distributed both via Kafka and email.
+
```
Remove headers for "Apache Kafka ..." and "GCN Notices Formats ...". Then move "This sample code ..." to after them, so that this is all background info for the sample code.

Vidushi-GitHub commented:
<img width="883" alt="Screenshot 2025-04-21 at 4 06 00 PM" src="https://github.com/user-attachments/assets/46980426-ef93-4a29-92cf-31fabfdc921b" />

lpsinger commented on app/routes/docs.sample.xml.md:
````diff
@@ -0,0 +1,45 @@
+Use the following to save the data to a local file:
+
+```python
+def save_alert(message):
+    with open('path/to/your/file', 'wb') as file:
+        file.write(message.value())
````
This is out of context. What is it supposed to do?

Vidushi-GitHub replied:
Ok.
XML content is taken as it is from the old document.

lpsinger commented on app/routes/docs.sample.xml.md:
```diff
@@ -0,0 +1,45 @@
+import xml.etree.ElementTree as ET
+import xmltodict
+
+def parse_text_alert_to_dict(message_value):
+    return dict(email.message_from_bytes(message_value))
```
This function has nothing to do with XML.

Vidushi-GitHub replied:
yes, it's related with email messages, either I need to merge it with xml or re-arrange somewhere else.

lpsinger commented on app/routes/docs.sample.kafka.md:
```diff
@@ -0,0 +1,121 @@
+---
+handle:
+  breadcrumb: Kafka Usage
```
Can you think of a less vague title than "Usage"?

jracusin replied:
Further down it says "Kafka Message Consumption".  Perhaps something like that would be more specific.

lpsinger replied:
How about "Consuming Kafka Messages?"

lpsinger commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,24 @@
+
+## Working with different GCN Notice Formats:
+
+- [Kafka Message Consumption](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
```
```suggestion
- [Consuming Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
```

lpsinger commented on app/routes/docs.sample.xml.md:
```diff
@@ -0,0 +1,54 @@
+
+## MIME-Encoded Messages
+
+The following function uses the `email` library to parse MIME-encoded messages.
```
This is the one and only place that we have ever described "MIME-encoded messages".

I think that this code is supposed to parse the GCN Classic text format.

Vidushi-GitHub commented:
<img width="823" alt="Screenshot 2025-05-02 at 5 42 38 PM" src="https://github.com/user-attachments/assets/3d010681-eda0-4b8e-884c-a5247c682ec4" />

Moved save alerts data to Kafka Page
<img width="884" alt="Screenshot 2025-05-02 at 5 43 20 PM" src="https://github.com/user-attachments/assets/f107d243-c0ab-449b-ad3f-9032ee17cf93" />

Moved Text messages handling to new page 
<img width="833" alt="Screenshot 2025-05-02 at 5 43 03 PM" src="https://github.com/user-attachments/assets/12155c73-afa0-47f3-9e61-88151d9e22d5" />

Vidushi-GitHub commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+  breadcrumb: TEXT
+---
+
+# GCN Classic Text Messages
```
```suggestion
# GCN Notices Emails
```

Vidushi-GitHub commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+  breadcrumb: TEXT
+---
+
+# GCN Notices Emails
```
```suggestion
# Plain Text Messages
```

jracusin commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+
+# Plain Text Messages
+
+GCN Classic Notices are also distributed as email messages. To extract useful information from these alerts, one can use Python's built-in [`email`](https://docs.python.org/3/library/email.html) library.
```
```suggestion
GCN Classic plain text-format Notices and GCN Circulars can be distributed as email messages. To extract useful information from these alerts, one can use Python's built-in [`email`](https://docs.python.org/3/library/email.html) library.
```

jracusin commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+  breadcrumb: TEXT
+---
+
+# Plain Text Messages
```
```suggestion
# Plain Text Format Notices
```

jracusin commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+---
+handle:
+  breadcrumb: TEXT
```
```suggestion
  breadcrumb: Plain Text
```

jracusin commented on app/routes/docs.sample.xml.md:
```diff
@@ -0,0 +1,32 @@
+  breadcrumb: XML
+---
+
+# XML Data
```
```suggestion
# VOEvent XML Data
```

jracusin commented on app/routes/docs.sample.xml.md:
```diff
@@ -0,0 +1,32 @@
+---
+handle:
+  breadcrumb: XML
```
```suggestion
  breadcrumb: VOEvent XML
```

jracusin commented on app/routes/docs.sample.xml.md:
```diff
@@ -0,0 +1,32 @@
+
+# XML Data
+
+Here is a collection of functions and example code that may be useful when working with XML and Multipurpose Internet Mail Extensions (MIME) formats.
```
```suggestion
Here is a collection of functions and example code that may be useful when working with [VOEvent XML](https://wiki.ivoa.net/internal/IVOA/IvoaVOEvent/voevent_v2.html) and Multipurpose Internet Mail Extensions (MIME) formats.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+## Working with different GCN Notice Formats:
+
+- [Consuming Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
+- [Text Messages](sample/text) demonstrates how to parse a GCN message.
```
```suggestion
- [Plain Text Notices](sample/text) demonstrates how to parse emailed plain text format Notices.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+
+- [Consuming Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
+- [Text Messages](sample/text) demonstrates how to parse a GCN message.
+- [XML Notices](sample/xml) provides guidance for interpreting and extracting data from XML formatted GCN Notices.
```
```suggestion
- [VOEvent XML Notices](sample/xml) provides guidance for interpreting and extracting data from VOEvent XML formatted GCN Notices.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+- [Text Messages](sample/text) demonstrates how to parse a GCN message.
+- [XML Notices](sample/xml) provides guidance for interpreting and extracting data from XML formatted GCN Notices.
+- [JSON Notices](sample/json) presents sample code for processing GCN Notices encoded in JSON format.
+- [HEALPix Sky Map](sample/healpix) explains how to interpret and utilize HEALPix format sky maps included in GCN data.
```
```suggestion
- [HEALPix Sky Map](sample/healpix) explains how to interpret and utilize HEALPix format sky maps included in GCN Notices, and instructions for Producers on how to encode FITS files in JSON format Notices.
```

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+- [Text Messages](sample/text) demonstrates how to parse a GCN message.
+- [VOEvent XML Notices](sample/xml) provides guidance for interpreting and extracting data from VOEvent XML formatted GCN Notices.
+- [JSON Notices](sample/json) presents sample code for processing GCN Notices encoded in JSON format.
+- [HEALPix Sky Map](sample/healpix) explains how to interpret and utilize HEALPix format sky maps included in GCN Notices, and instructions for Producers on how to encode FITS files in JSON format Notices.
```
```suggestion
- [HEALPix Sky Map](sample/healpix) explains how to interpret and utilize HEALPix format sky maps included in GCN Notices.
```

Vidushi-GitHub commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+- [Consuming Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
+- [Text Messages](sample/text) demonstrates how to parse a GCN message.
+- [VOEvent XML Notices](sample/xml) provides guidance for interpreting and extracting data from VOEvent XML formatted GCN Notices.
+- [JSON Notices](sample/json) presents sample code for processing GCN Notices encoded in JSON format.
```
```suggestion
- [JSON Notices](sample/json) presents sample code for processing GCN Notices encoded in JSON format, and instructions for Producers on how to encode FITS files in JSON format Notices.
```

Vidushi-GitHub commented:
<img width="880" alt="Screenshot 2025-05-06 at 3 04 52 PM" src="https://github.com/user-attachments/assets/05b7969a-ad6f-48f4-aee3-a9cab06ef002" />
<img width="832" alt="Screenshot 2025-05-06 at 3 05 02 PM" src="https://github.com/user-attachments/assets/c8afd6b0-cc34-4af3-afd5-6da46bb97ca9" />

dakota002 commented on app/routes/docs.tsx:
```diff
@@ -155,6 +155,31 @@ export default function () {
+                      JSON
+                    </NavLink>,
+                    <NavLink key="healpix" to="sample/healpix">
+                      HEALPix                    
```
```suggestion
                      HEALPix                    
                    </NavLink>,
```

Vidushi-GitHub committed:
feedback

Vidushi-GitHub commented:
<img width="845" alt="Screenshot 2025-05-06 at 6 52 43 PM" src="https://github.com/user-attachments/assets/009dc57e-8d3b-4e30-a8f1-cc4680318293" />
<img width="841" alt="Screenshot 2025-05-06 at 6 52 52 PM" src="https://github.com/user-attachments/assets/ecf18753-458f-4e2e-aaad-45c37e2a4e45" />

Vidushi-GitHub commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+## Working with different GCN Notice Formats:
+
+- [Consuming Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
+- [Plain Text Notices & Circulars](sample/text) demonstrates how to parse emailed plain text format GCN messages.
```
Emailed plain text format is applicable for Circulars as well.

lpsinger replied:
If they are, then it's a coincidence. Please don't mention Circulars here. This is irrelevant.

Please don't conflate the GCN Classic text format with the email delivery method.

Vidushi-GitHub committed:
Matching Page Title and Level 1 Heading

improved level 1 headings

Vidushi-GitHub commented:
<img width="845" alt="Screenshot 2025-05-08 at 5 17 56 PM" src="https://github.com/user-attachments/assets/6aa1625d-7e23-4c83-a861-a042eb4fefff" />

jracusin commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+
+## Working with different GCN Notice Formats
+
+- [Consuming Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
```
```suggestion
- [Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
```

Vidushi-GitHub committed:
Update app/routes/docs.sample._index.mdx

Co-authored-by: Judy Racusin <judith.racusin@gmail.com>

lpsinger commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+---
+handle:
+  breadcrumb: Plain Text Emails
+---
+
+# Plain Text Emails
```
```suggestion
  breadcrumb: Plain Text
---

# Plain Text
```

lpsinger commented on app/routes/docs.tsx:
```diff
@@ -155,8 +155,34 @@ export default function () {
+                      Kafka Messages
+                    </NavLink>,
+                    <NavLink key="text" to="sample/text">
+                      Plain Text Emails
```
```suggestion
                      Plain Text
```

Vidushi-GitHub committed:
Update app/routes/docs.tsx

Co-authored-by: Leo Singer <leo.p.singer@nasa.gov>

Vidushi-GitHub committed:
Update app/routes/docs.sample.text.md

Co-authored-by: Leo Singer <leo.p.singer@nasa.gov>

Vidushi-GitHub committed:
Update app/routes/docs.sample._index.mdx

Co-authored-by: Leo Singer <leo.p.singer@nasa.gov>

lpsinger commented on app/routes/docs.sample._index.mdx:
```diff
@@ -0,0 +1,25 @@
+## Working with different GCN Notice Formats
+
+- [Kafka Messages](sample/kafka) demonstrates how to consume and manage GCN messages using Kafka clients.
+- [Plain Text Notices](sample/text) demonstrates how to parse emailed plain text format GCN messages.
```
```suggestion
- [Plain Text Notices](sample/text) demonstrates how to parse plain text format GCN messages.
```

lpsinger commented on app/routes/docs.sample.text.md:
```diff
@@ -0,0 +1,17 @@
+
+# Plain Text
+
+GCN Classic Notices are also distributed as email messages. To extract header information, one can use Python's built-in [`email`](https://docs.python.org/3/library/email.html) library.
```
```suggestion
GCN Classic Notices are also distributed as plain text. To extract header information, one can use Python's built-in [`email`](https://docs.python.org/3/library/email.html) library.
```

Vidushi-GitHub committed:
Update app/routes/docs.sample._index.mdx

Co-authored-by: Leo Singer <leo.p.singer@nasa.gov>

Vidushi-GitHub committed:
Update app/routes/docs.sample.text.md

Co-authored-by: Leo Singer <leo.p.singer@nasa.gov>
